### PR TITLE
[FLINK-3680][web frontend] Remove "(not set)" text in the Job Plan UI

### PR DESCRIPTION
**Pull request by mushketyk**
This PR removes the "(not set)" text in the web frontend.

Currently it looks like this: https://issues.apache.org/jira/secure/attachment/12796006/Screen%20Shot%202016-03-29%20at%208.13.12%20PM.png

Now it looks like: 
fixed labels
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)
- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added
- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by mushketyk** [comment]
@iampeter 
Did you have a chance to take a look at this change?

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java [hunk 1]
```diff
@@ -52,10 +52,10 @@ public static String generatePlan(JobGraph jg) {
 			for (JobVertex vertex : jg.getVertices()) {
 				
 				String operator = vertex.getOperatorName() != null ? 
-						vertex.getOperatorName() : NOT_SET;
+						vertex.getOperatorName() : "";
```
Comment (greghogan) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java]: Why not change `NOT_SET` to be `""`?

**Comment by mushketyk** [comment]
Hi @greghogan 
I've fixed the code according to your suggestion.
Could you give it another look?

File: flink-runtime-web/web-dashboard/app/scripts/modules/jobs/jobs.dir.coffee [hunk 2]
```diff
@@ -281,7 +281,8 @@ angular.module('flinkApp')
         # Otherwise add infos    
         labelValue += "<h5>" + info + " Node</h5>"  if isSpecialIterationNode(info)
         labelValue += "<h5>Parallelism: " + el.parallelism + "</h5>"  unless el.parallelism is ""
-        labelValue += "<h5>Operation: " + shortenString(el.operator_strategy) + "</h5>"  unless el.operator is `undefined`
+        if el.operator_strategy
+          labelValue += "<h5>Operation: " + shortenString(el.operator_strategy) + "</h5>"  unless el.operator is `undefined`
```
Comment (iampeter) [inline flink-runtime-web/web-dashboard/app/scripts/modules/jobs/jobs.dir.coffee]: It's fine, but maybe:
```
labelValue += "<h5>Operation: " + shortenString(el.operator_strategy) + "</h5>"  unless !el.operator_strategy || el.operator is `undefined`
```

doesn't really matter much, can be as well as you've put it.

Comment (mushketyk) [inline flink-runtime-web/web-dashboard/app/scripts/modules/jobs/jobs.dir.coffee]: I think this will make it more consistent. I'll update it.

**Comment by mushketyk** [comment]
Hi @iampeter 
I've updated the PR according to you suggestion.

**Comment by iampeter** [comment]
@mushketyk from my side looks good

**Comment by greghogan** [comment]
1) `JsonPlanGenerator.java:55` should be reverted to use `NOT_SET`.
2) `index.js` is changing jobServer from the empty string to 'http://localhost:8081/'.
3) `vendor.js` is upgrading the version of `imagesLoaded` from 4.1.0 to 4.1.1. In the past we have specified in `flink-runtime-web/web-dashboard/bower.json` to not upgrade point releases since the changes were obfuscating the remainder of the pull request. I have not noticed if we have subsequently followed up with a PR to upgrade the libraries. We may be at a good point in the 1.2 release cycle to do this now.

**Comment by iampeter** [comment]
@greghogan right - missed the generated files. @mushketyk I suppose `gulp`ing will help

**Comment by mushketyk** [comment]
@iampeter @greghogan I've update the PR according to your review.

**Comment by uce** [comment]
Would be cool if we could also merge this to `release-1.1` if it is good to merge.

**Comment by greghogan** [comment]
Merging this ...

**Pull request closed without merging**